monitors/archive_monitor/isis_archive_monitor_win_service.py

```diff
@@ -0,0 +1,61 @@
+"""
+Holds the class for overidding QueueService framework
+"""
+# Can't import on travis (linux) server
+# pylint:disable=import-error
+import sys
+import time
+
+import servicemanager
+import win32event
+import win32serviceutil
+
+from monitors.windows_service import WindowsService
+from monitors.archive_monitor.isis_archive_monitor import ArchiveMonitor
+from monitors.archive_monitor.isis_archive_monitor_helper import SLEEP_TIME
+
+
+class PollService(WindowsService):
+    """
+    Class to override windows ServiceFramework
+    """
+    _svc_name_ = "AutoReductionArchiveMonitor"
+    _svc_display_name_ = "Autoreduce Archive Monitor"
+    _svc_description_ = "Polls ISIS data archive directory to check for new files"
+
+    # pylint: disable=invalid-name
+    def SvcDoRun(self):
+        """
+        Run the main function in isis_archive_monitor which starts the polling process
+        """
+        servicemanager.LogMsg(servicemanager.EVENTLOG_INFORMATION_TYPE,
+                              servicemanager.PYS_SERVICE_STARTED,
+                              (self._svc_name_, ''))
+        monitor = ArchiveMonitor("POLARIS")
+        while 1:
+            monitor.perform_check()
+            sleep_timer = 0
+            end_service_loop = False
+            while sleep_timer < SLEEP_TIME:
+                time_to_sleep = 30 if SLEEP_TIME > 30 else SLEEP_TIME
```

Comment:
Why 30? Could we encapsulate this magic value?

Reply:
I believe this is the max length of time a windows will try and stop a service for before it times out but you're right we should make this clearer!